### PR TITLE
Merge the changes made to 3.x to master

### PR DESCRIPTION
**Pull request by metmajer**
Over the recent months, changes have been made to a long-running feature branch which resulted in the current 3.x branch. Now, these changes are merged back into master \*imagine scary music playing\*.

I had feared that this merge would get us into merge hell, but it was much less painful than I thought. After looking through the resulting changes line by line, I applied a few fix commits at the end of the history. Regarding the overall history, doing multiple cherry picks would have created a cleaner one, but I am inclined to stashing this entire branch into a single commit before merging into master. Thoughts?

FYI @jorge-romero @jafarre-viewnext @s2oBCN @segator @michaelsauter @clemensutschig

### CODE REVIEW

**Comment by metmajer** [comment]
@renedupont please double-check if your changes to Antora are properly reflected, thanks

**Review by jafarre-viewnext — APPROVED** [review]
I agree about squashing the commits.

File: src/org/ods/orchestration/DeployStage.groovy [hunk 1]
```diff
@@ -36,7 +37,8 @@ class DeployStage extends Stage {
             // not work on agent nodes yet.
             if (agentPodCondition) {
                 script.node {
-                    script.sh "cp -r ${standardWorkspace}/docs ${script.env.WORKSPACE}/docs"
+                    script.sh "cp -r ${standardWorkspace}/docs ${script.env.WORKSPACE}"
+                    script.sh "cp -r ${standardWorkspace}/projectData ${script.env.WORKSPACE}"
```
Comment (michaelsauter) [inline src/org/ods/orchestration/DeployStage.groovy:41]: Are these path changes intentional?

**Review by michaelsauter — COMMENTED** [review]
Cool! Looks good except the question I have on the path changes ...

File: src/org/ods/orchestration/DeployStage.groovy (hunk 1 above)
Comment (metmajer) [reply src/org/ods/orchestration/DeployStage.groovy:41]: @michaelsauter I asked myself the same question. With `cp -r`, a copy operation from `$pathA/$dirA` to `$pathB/$dirA` is be redundant. Therefore, you can also write it as `cp -R $pathA/$dirA $pathB`. `$dirA` is nevertheless created inside `$pathB` unless the source would have a trailing slash, as in `$pathA/$dirA/`.

**Comment by renedupont** [comment]
> 
> 
> @renedupont please double-check if your changes to Antora are properly reflected, thanks

Not entirely sure what changes you are referring to, in case you are talking about https://github.com/opendevstack/ods-jenkins-shared-library/blob/3.x-to-master/docs/antora.yml#L3 I don't see an issue here. It is already like this on master.

**Comment by metmajer** [comment]
It's a wrap, @jafarre-viewnext @felipecruz91 @martsec @georgfedermann @hrcornejo @jorge-romero @s2oBCN @nichtraunzer @frankjoas @renedupont @michaelsauter @gerardcl @clemensutschig @stitakis. Thanks! 💯

**Pull request merged**